apps/nextjs-example/pages/index.tsx

```diff
@@ -340,6 +368,109 @@ function OptionalFunctionality() {
         );
     };
 
+    const onSubmitScript = async () => {
+        const payload = {
+            type: "script_payload",
+            code: {
+                bytecode: DO_NOTHING_SCRIPT
+            },
+            type_arguments: [],
+            arguments: []
+        };
+
+        const response = await signAndSubmitTransaction(payload);
+        try {
+            if (response?.hash === undefined) {
+                throw new Error(`No response given ${response}`)
+            }
+            await aptosClient(network?.name.toLowerCase()).waitForTransaction(response.hash);
```

Comment:
Mmm ok at least what we're doing on Petra is, we're submitting the transaction and waiting for its result.
If there's an error when submitting, we show it in the prompt, but the prompt stays open.

Maybe we should enforce the same for other wallets?
Otherwise then it becomes the dapp's responsibility to display the transaction error, and we have to figure out passing just the right details back to the dapp.

Reply:
I've removed the script functionality for now.

This is up to the wallet, but I think we either way have to use waitForTransaction to deal with eventual consistency (the wallet may use a different endpoint)